tox.ini

```diff
@@ -1,46 +1,48 @@
 [tox]
-envlist = {py35,py36,py37}-{cov,nocov,diffcov},qa,docs
+envlist = {py36,py37,py38,py39}-{cov,nocov,diffcov}, qa, docs
 skip_missing_interpreters = True
 
 [testenv]
 commands =
+    python -c "import sys; print(sys.version); print(sys.executable)"
+    python -c "import os,pprint; os.makedirs('_dynamic', exist_ok=True); pprint.pprint(dict(os.environ), stream=open('_dynamic/ENV."{envname}"', 'wt'))"
     nocov: python -m nose2 -v {posargs}
-    {cov,diffcov}: python -m coverage run {[coverage]rc} -m nose2
-    {cov,diffcov}: python -m coverage combine {[coverage]rc}
-    cov: python -m coverage html {[coverage]rc}
-    cov: python -m coverage report -m {[coverage]rc} --fail-under=100
-    diffcov: python -m coverage xml {[coverage]rc}
+    {cov,diffcov}: python -m coverage run -m nose2
+    {cov,diffcov}: python -m coverage combine
+    cov: python -m coverage html --directory=htmlcov/{envname}
+    cov: python -m coverage report -m --fail-under=100
+    diffcov: python -m coverage xml
     diffcov: diff-cover coverage.xml --html-report diffcov.html
     diffcov: diff-cover coverage.xml --fail-under=100
+    profile: python -m nose2 --profile {posargs}
+    profile: python -c "open('_dynamic/cPROFILE."{envname}"', 'wb').write(open('_dynamic/cPROFILE', 'rb').read())"
+    python -c "import sys; print(sys.version); print(sys.executable)"
 #sitepackages = True
 usedevelop = True
 deps =
     nose2
-    flufl.testing
     {cov,diffcov}: coverage
     diffcov: diff_cover
 setenv =
-    cov: COVERAGE_PROCESS_START={[coverage]rcfile}
+    cov: COVERAGE_PROCESS_START=.coveragerc
     cov: COVERAGE_OPTIONS="-p"
     cov: COVERAGE_FILE={toxinidir}/.coverage
-    py35: INTERP=py35
     py36: INTERP=py36
     py37: INTERP=py37
+    py38: INTERP=py38
+    py39: INTERP=py39
+    pypy3: INTERP=pypy3
+    nocov: PYTHONASYNCIODEBUG=1
     PLATFORM={env:PLATFORM:linux}
 passenv =
     PYTHON*
 
-[coverage]
-rcfile = {toxinidir}/.coverage.ini
-rc = --rcfile={[coverage]rcfile}
-
 [testenv:qa]
 basepython = python3
 commands =
     python -m flake8 aiosmtpd
 deps =
     flake8
-    flufl.testing
```

Comment:
Conflicts with `nose2.plugins.prof` and [no longer being maintained anyways](https://gitlab.com/warsaw/flufl.testing/-/tree/master/):

> As of 2020-06-17 I have archived this project and will no longer be
developing it.  This library's flake8 functionality is obsoleted by isort, and
pytest has clearly won hearts and minds over nose2.